prow/kube/client.go

```diff
@@ -195,6 +198,72 @@ func NewClientInCluster(namespace string) (*Client, error) {
 	}, nil
 }
 
+// Cluster represents the information necessary to talk to a Kubernetes
+// master endpoint.
+type Cluster struct {
+	// The IP address of the cluster's master endpoint.
+	Endpoint string `json:"endpoint"`
+	// Base64-encoded public cert used by clients to authenticate to the
+	// cluster endpoint.
+	ClientCertificate string `json:"clientCertificate"`
+	// Base64-encoded private key used by clients..
+	ClientKey string `json:"clientKey"`
+	// Base64-encoded public certificate that is the root of trust for the
+	// cluster.
+	ClusterCACertificate string `json:"clusterCaCertificate"`
+}
+
+// NewClientFromFile reads a Cluster object at clusterPath and returns an
+// authenticated client using the keys within.
+func NewClientFromFile(clusterPath, namespace string) (*Client, error) {
+	data, err := ioutil.ReadFile(clusterPath)
+	if err != nil {
+		return nil, err
+	}
+	var c Cluster
+	if err := json.Unmarshal(data, &c); err != nil {
+		return nil, err
+	}
+	return NewClient(&c, namespace)
+}
+
+// NewClient returns an authenticated Client using the keys in the Cluster.
+func NewClient(c *Cluster, namespace string) (*Client, error) {
```

Comment:
It's certainly possible but it's a lot of work grumble grumble also a test server won't be identical to a k8s api server grumble grumble fine.